### ISSUE TITLE
Eslint reporting errors but not being populates into Location list

### ISSUE THREAD

**Issue by zacbraddy** [vim-syntastic/syntastic#2155]
Hi all

I've spent some time now googling and tinkering trying to get my location list to populate with the errors I'm getting in eslint. Syntactic is able to see my path variable using the `syntactic#utils#system` command and the version of eslint I'm using is 3.19.0 (this is necessary as I'm using the global install hack to make syntastic able to use eslint, I don't expect the eslint version to be the problem because I have this working on my machine at work and I'm trying to set up the same thing at home). I have been playing for some time with different settings and here is the list of settings I am using which I believe to be correct:
```
let g:syntastic_always_populate_loc_list = 1
let g:syntastic_loc_list_height = 5
let g:syntastic_auto_loc_list = 0
let g:syntastic_check_on_open = 1
let g:syntastic_check_on_wq = 1
let g:syntastic_javascript_checkers = ['eslint']
let g:syntastic_javascript_eslint_exe = 'eslint --'
let g:syntastic_javascript_eslint_generic = 1
let g:syntastic_debug=3
set noshellslash
```

I am on a windows machine doing this. Obviously the debug is just on until I can work out what is going on so that's not all that relevant.

I've made a test.js file in my project folder which simply has `x=1` in it and when I run `eslint -- Frontend/test.js` as syntastic does I get the error back.

Here is what I get back from my SyntasticInfo
```
Syntastic version: 3.8.0-94 (Vim 800, Windows)
Info for filetype: javascript.jsx
Global mode: active
Filetype javascript.jsx is active
The current file will be checked automatically
Available checker: javascript/eslint
Currently enabled checker: javascript/eslint
syntastic: 60.668916: &shell = 'cmd' (!), &shellcmdflag = '/c', &shellpipe = '>%s 2>&1', &shellquote = '', &shellredir = '>%s 2>&1', &shelltemp = 1, &shellxquote = '', &autochdir = 0, &shellslash = 0, &shellxescape = '"&|<>()@^'
```

And here is my `:mes` after opening `test.js`
```
syntastic: 10.543493: g:syntastic_version = '3.8.0-94 (Vim 800, Windows)'
syntastic: 10.543949: &shell = 'cmd' (!), &shellcmdflag = '/c', &shellpipe = '>%s 2>&1', &shellquote = '', &shellredir = '>%s 2>&1', &shelltemp = 1, &shellxquote = '', &autochdir = 0, &shellslash = 0, &shellxescape = '"&|<>()@^'
syntastic: 10.544655: UpdateErrors (auto): default checkers
syntastic: 10.545142: CacheErrors: default checkers
syntastic: 10.545814: g:syntastic_aggregate_errors = 0
syntastic: 10.546089: getcwd() = 'D:\Projects'
syntastic: 10.556955: CacheErrors: Invoking checker: javascript/eslint
syntastic: 10.557580: SyntasticMake: called with options: {'postprocess': ['guards'], 'errorformat': '%E%f: line %l\, col %c\, Error - %m,%W%f: line %l\, col %c\, Warning - %m', 'makeprg': 'eslint -- "Frontend\test.js"'}
syntastic: 11.650892: system: command run in 1.092504s
syntastic: 11.651197: checker output: ['', 'D:\Projects\Frontend\test.js', '  1:1  error    ''x'' is not defined  no-undef', '  1:4  warning  Missing semicolon   semi', '', '✖ 2 problems (1 error, 1 warning)', '', '']
syntastic: 11.651915: raw loclist: [{'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'D:\Projects\Frontend\test.js'}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': '  1:1  error    ''x'' is not defined  no-undef'}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': '  1:4  warning  Missing semicolon   semi'}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': '✖ 2 problems (1 error, 1 warning)'}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}]
syntastic: 11.654430: postprocess: [{'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'D:\Projects\Frontend\test.js'}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': '  1:1  error    ''x'' is not defined  no-undef'}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': '  1:4  warning  Missing semicolon   semi'}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': '✖ 2 problems (1 error, 1 warning)'}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}]
syntastic: 11.655958: getLocList: checker javascript/eslint returned 1
syntastic: 11.656159: javascript/eslint raw: [{'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'D:\Projects\Frontend\test.js'}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': '  1:1  error    ''x'' is not defined  no-undef'}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': '  1:4  warning  Missing semicolon   semi'}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': '✖ 2 problems (1 error, 1 warning)'}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}, {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}]
syntastic: 11.657671: quiet_messages filter: {}
syntastic: 11.657874: getLocList: checker javascript/eslint run in 1.100622s
syntastic: 11.658168: aggregated: {'_sorted': 0, '_name': '', '_owner': 1, '_columns': 1, '_rawLoclist': []}
```

You can see that my error is comming through in that output but the getLocList is returning with an error code 1 which for the life of me I can't find the meaning of.

I'm really sorry to waste people's time troubleshooting but I'm totally at a loss here and I'm hoping someone might be able to point out something silly that I'm missing. Any help offered would be greatly appreciated.

Thanks in advance
Zac

**Comment by zacbraddy** [comment]
Worked it out. Turns out the location list is expecting the output in compact format so I had to add a `-f compact` to the end of the my `g:syntastic_javascript_eslint_exe` variable so it looked like `eslint -f compact --`

**Comment by lcd047** [comment]
You can save a lot of time by spending a few minutes to understand what you're doing instead of trying random solutions from Google to problems you don't have.  In particular you could have just left out these lines:
```vim
let g:syntastic_javascript_eslint_exe = 'eslint --'
let g:syntastic_javascript_eslint_generic = 1
```
You ended up just reconstructing the defaults.

**Comment by zacbraddy** [comment]
I totally agree with your sentiment. In this case I'm rather new to vim and so I'm learning a lot daily. My real problem in these situations is that I have unknown unknowns which faces me with two possible paths; I either need to learn about every option of every plugin until I find my problem or, what I did here, google solutions and try to learn what each solution is doing until I can come up with my own. 6 hours after starting this endeavor I feel like I have a much better understanding of the solution I landed on (it didn't stop after I made my last comment ;) ). Sorry to have wasted your time but I really appreciate you taking the time to read and reply to my issue.
